packages/webpack/lib/utils/compiler.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+const { join } = require('path');
+const webpack = require('webpack');
+const MemoryFS = require('memory-fs');
+const sourceMapSupport = require('source-map-support');
+
+function compilerHandler(config, callback, compileError, stats) {
```

Comment:
I am really not too keen on this API, sorry! If we had to keep things callback-style, why not stick to the tried-and-tested Node-style callbacks (`(err, res) => {}`)?

Generally, I would prefer this thing to directly returning an `EnhancedPromise`, if at all possible, instead of relying on custom data structures and magic strings.